tests/post: Reduce some flakiness

gdemonet opened the pull request:
**docs, tests: Apply specific '.deployed' states**

When changing the CSC ConfigMap of a component we were often running
the whole `metalk8s.deployed` state.
We now document which state to run for each specific component, and
ensure the tests are following this documentation.

---

**tests: Clean-up Dex-related tests**

Mainly remove the `request_retry_session` fixture and replace it with
direct calls to `utils.request_retry_session` when needed, so one can
adjust the expected error codes and the number of retries depending on
the use-case.
Adding `502` and `504` to the list of expected error codes, and bumping
retries from 3 to 5, for all requests specifically targeted at Dex.

---

**tests: Wait for rollout of CSC changes to complete**

Not waiting was leading to some flaky issues where the expected changes
were not yet propagated to the tested service, hence breaking the test.

alexandre-allard committed:
docs, tests: Apply specific '.deployed' states

When changing the CSC ConfigMap of a component we were often running
the whole `metalk8s.deployed` state.
We now document which state to run for each specific component, and
ensure the tests are following this documentation.

bert-e commented:
# Hello gdemonet,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @thomasdanan

* @alexandre-allard-scality

* @jbwatenbergscality

* @gdemonet

* @lucieleonard

* @chengyanjin

* @teddyandrieux

* @nicolast

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/reduce-service-config-test-flakiness` into
`development/2.8`
* [w/2.9/bugfix/reduce-service-config-test-flakiness](https://github.com/scality/metalk8s/compare/development/2.9...w/2.9/bugfix/reduce-service-config-test-flakiness) will be merged into `development/2.9` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @thomasdanan

* @alexandre-allard-scality

* @jbwatenbergscality

* @gdemonet

* @lucieleonard

* @chengyanjin

* @teddyandrieux

* @nicolast

gdemonet commented:
Actually, @alexandre-allard-scality has a better solution: 876ebcc

The proposed approach here encodes in the tests the state which should be executed, whereas the other approach makes it part of "the feature", and thus assumes a user would run it.
I will cherry-pick his commit and also update the documentation in the relevant places.

bert-e commented:
# History mismatch



Merge commit #0172a2994f8119a4d99e9eeb9a35b19a16505e2e on the integration branch
`w/2.9/bugfix/reduce-service-config-test-flakiness` is merging a branch which is neither the current
branch `bugfix/reduce-service-config-test-flakiness` nor the development branch
`development/2.9`.

It is likely due to a rebase of the branch `bugfix/reduce-service-config-test-flakiness` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

gdemonet commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/reduce-service-config-test-flakiness` into
`development/2.8`
* [w/2.9/bugfix/reduce-service-config-test-flakiness](https://github.com/scality/metalk8s/compare/development/2.9...w/2.9/bugfix/reduce-service-config-test-flakiness) will be merged into `development/2.9` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @thomasdanan

* @alexandre-allard-scality

* @jbwatenbergscality

* @gdemonet

* @lucieleonard

* @chengyanjin

* @teddyandrieux

* @nicolast

alexandre-allard commented on docs/operation/cluster_and_service_configuration.rst:
```diff
@@ -610,7 +610,7 @@ To change the number of replicas, perform the following operations:
       root\@bootstrap $ kubectl exec --kubeconfig /etc/kubernetes/admin.conf \\
                          -n kube-system -c salt-master salt-master-bootstrap \\
-                         -- salt-run state.sls metalk8s.deployed \\
+                         -- salt-run state.sls metalk8s.logging.deployed \\
```
The `Replicas Count Customization` section is not only about Loki, but for all services using the CSC mechanism.

gdemonet replied:
Ah right, went too fast. Good catch :ok_hand:

gdemonet replied:
Done

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @thomasdanan

* @alexandre-allard-scality

* @jbwatenbergscality

* @gdemonet

* @lucieleonard

* @chengyanjin

* @teddyandrieux

* @nicolast





The following reviewers are expecting changes from the author, or must review again:

* @alexandre-allard-scality

gdemonet committed:
tests: Clean-up Dex-related tests

Mainly remove the `request_retry_session` fixture and replace it with
direct calls to `utils.request_retry_session` when needed, so one can
adjust the expected error codes and the number of retries depending on
the use-case.
Adding `502` and `504` to the list of expected error codes, and bumping
retries from 3 to 5, for all requests specifically targeted at Dex.

bert-e commented:
# History mismatch



Merge commit #55a72a4ca156cdac6c0b70371045b6c3e9ddbbe8 on the integration branch
`w/2.9/bugfix/reduce-service-config-test-flakiness` is merging a branch which is neither the current
branch `bugfix/reduce-service-config-test-flakiness` nor the development branch
`development/2.9`.

It is likely due to a rebase of the branch `bugfix/reduce-service-config-test-flakiness` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

gdemonet commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.9/bugfix/reduce-service-config-test-flakiness` with contents from `bugfix/reduce-service-config-test-flakiness`
and `development/2.9`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.9/bugfix/reduce-service-config-test-flakiness origin/development/2.9
 $ git merge origin/bugfix/reduce-service-config-test-flakiness
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.9/bugfix/reduce-service-config-test-flakiness
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @thomasdanan

* @alexandre-allard-scality

* @jbwatenbergscality

* @gdemonet

* @lucieleonard

* @chengyanjin

* @teddyandrieux

* @nicolast

gdemonet commented:
/approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @thomasdanan

* @alexandre-allard-scality

* @jbwatenbergscality

* @gdemonet

* @lucieleonard

* @chengyanjin

* @teddyandrieux

* @nicolast










*The following options are set:* **approve**

gdemonet commented:
Still one flaky occurrence to fix on the service-configuration test:

```
=================================== FAILURES ===================================
________________ test_static_user_change[paramiko://bootstrap] _________________
[...]
        assert auth_response.text is not None
        if should_fail:
>           assert auth_response.status_code == 200
E           assert 303 == 200
E            +  where 303 = <Response [303]>.status_code
```

This is likely due to a race condition:

- the test applies a service-configuration change to edit the password of our static user in Dex
- the test then attempts to login using the old password, and expects this to fail
- however, it succeeds, indicating that the login attempt occurred before the changes were fully propagated to Dex

This race issue directly comes from Kubernetes declarative approach: our states only ensure that the declaration of desired state is updated, not that the observed state has been reconciled to match this desired state declaration.

I will work around this by adding a `kubectl rollout status deploy/dex -n metalk8s-auth`.

bert-e commented:
# Branches have diverged



This pull request's source branch `bugfix/reduce-service-config-test-flakiness` has diverged from
`development/2.8` by more than 50 commits.

To avoid any integration risks, please re-synchronize them using one of the
following solutions:

* Merge `origin/development/2.8` into `bugfix/reduce-service-config-test-flakiness`
* Rebase `bugfix/reduce-service-config-test-flakiness` onto `origin/development/2.8`

Note: If you choose to rebase, you may have to ask me to rebuild
integration branches using the `reset` command.

gdemonet committed:
tests: Wait for rollout of CSC changes to complete

Not waiting was leading to some flaky issues where the expected changes
were not yet propagated to the tested service, hence breaking the test.

bert-e commented:
# History mismatch



Merge commit #239f9b7a65c674ea10f4cb112a8f70844661415f on the integration branch
`w/2.9/bugfix/reduce-service-config-test-flakiness` is merging a branch which is neither the current
branch `bugfix/reduce-service-config-test-flakiness` nor the development branch
`development/2.9`.

It is likely due to a rebase of the branch `bugfix/reduce-service-config-test-flakiness` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @nicolast

* @chengyanjin

* @alexandre-allard-scality

* @thomasdanan

* @gdemonet

* @teddyandrieux

* @lucieleonard

* @jbwatenbergscality

bert-e commented:
# History mismatch



Merge commit #07462431cf100677bca7558341d13836724a580a on the integration branch
`w/2.9/bugfix/reduce-service-config-test-flakiness` is merging a branch which is neither the current
branch `bugfix/reduce-service-config-test-flakiness` nor the development branch
`development/2.9`.

It is likely due to a rebase of the branch `bugfix/reduce-service-config-test-flakiness` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

bert-e commented:
# History mismatch



Merge commit #17803dc1238c9687905cc852fde3ab40471d04c4 on the integration branch
`w/2.9/bugfix/reduce-service-config-test-flakiness` is merging a branch which is neither the current
branch `bugfix/reduce-service-config-test-flakiness` nor the development branch
`development/2.9`.

It is likely due to a rebase of the branch `bugfix/reduce-service-config-test-flakiness` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

gdemonet commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/2.9/bugfix/reduce-service-config-test-flakiness` with contents from `bugfix/reduce-service-config-test-flakiness`
and `development/2.9`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/2.9/bugfix/reduce-service-config-test-flakiness origin/development/2.9
 $ git merge origin/bugfix/reduce-service-config-test-flakiness
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.9/bugfix/reduce-service-config-test-flakiness
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @nicolast

* @chengyanjin

* @alexandre-allard-scality

* @thomasdanan

* @gdemonet

* @teddyandrieux

* @lucieleonard

* @jbwatenbergscality

gdemonet commented:
/approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @nicolast

* @chengyanjin

* @alexandre-allard-scality

* @thomasdanan

* @gdemonet

* @teddyandrieux

* @lucieleonard

* @jbwatenbergscality










*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.8`

* :heavy_check_mark: `development/2.9`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.8`

* :heavy_check_mark: `development/2.9`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`



Please check the status of the associated issue None.

Goodbye gdemonet.
